src/app/eventsDecoder.js

```diff
@@ -0,0 +1,82 @@
+'use strict'
+var $ = require('jquery')
+var ethJSABI = require('ethereumjs-abi')
+
+/**
+  * Register to txListener and extract events
+  *
+  */
+class EventsDecoder {
+  constructor (opt = {}) {
+    this.txListener = opt.txListener
+    this.resolvedEvents = {}
+  }
+
+/**
+  * use Transaction Receipt to decode logs. assume that the transaction as already been resolved by txListener.
+  * logs are decoded only if the contract if known by remix.
+  *
+  * @param {Object} tx - transaction object
+  * @param {Function} cb - callback
+  */
+  parseLogs (tx, contractName, compiledContracts, cb) {
+    this.txListener.resolveTransactionReceipt(tx, (error, receipt) => {
+      if (error) cb(error)
+      this._decodeLogs(tx, receipt, contractName, compiledContracts, cb)
+    })
+  }
+
+  eventsOf (hash) {
+    return this.resolvedEvents[hash]
+  }
+
+  _decodeLogs (tx, receipt, contract, contracts, cb) {
+    if (!contract || !receipt.logs) {
+      return cb()
+    }
+    this._decodeEvents(tx, receipt.logs, contract, contracts, cb)
+  }
+
+  _eventABI (contractName, compiledContracts) {
+    var contractabi = JSON.parse(compiledContracts[contractName].interface)
+    var eventABI = {}
+    $.each(contractabi, function (i, funABI) {
```

Comment:
is `contractabi` an array?  ...and is this the only place where `jquery` is used?
if so ... you could get rid of the dependency by replacing it with a native
```js
contractabi.forEach(function (funABI, i) {
  // ... do stuff ...
})